To: vim-dev@vim.org
Subject: Patch 7.3.001
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.001
Problem:    When editing "src/main.c" and 'path' set to "./proto",
	    ":find e<C-D" shows ./proto/eval.pro instead of eval.pro.
Solution:   Check for path separator when comparing names. (Nazri Ramliy)
Files:	    src/misc1.c


*** ../vim-7.3.000/src/misc1.c	2010-08-15 21:57:27.000000000 +0200
--- src/misc1.c	2010-08-16 20:43:25.000000000 +0200
***************
*** 9317,9323 ****
  	    continue;  /* it's different when it's shorter */
  
  	rival = other_paths[j] + other_path_len - candidate_len;
! 	if (fnamecmp(maybe_unique, rival) == 0)
  	    return FALSE;  /* match */
      }
  
--- 9317,9324 ----
  	    continue;  /* it's different when it's shorter */
  
  	rival = other_paths[j] + other_path_len - candidate_len;
! 	if (fnamecmp(maybe_unique, rival) == 0
! 		&& (rival == other_paths[j] || vim_ispathsep(*(rival - 1))))
  	    return FALSE;  /* match */
      }
  
*** ../vim-7.3.000/src/version.c	2010-08-15 21:57:25.000000000 +0200
--- src/version.c	2010-08-16 20:53:09.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     1,
  /**/


-- 
From "know your smileys":
 (:-#	Said something he shouldn't have

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\        download, build and distribute -- http://www.A-A-P.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
